File: src/pl/Symbol.java
package pl;

public enum Symbol {
    NUMBER,
    LP,
    RP,
    PLUS,
    MINUS,
    MULTIPLY,
    DIVIDE,
    UNKNOWN;

    public static Symbol getSymbol(char c) {
        switch (c) {
            case '0':
            case '1':
            case '2':
            case '3':
            case '4':
            case '5':
            case '6':
            case '7':
            case '8':
            case '9':
                return NUMBER;
            case '(':
                return LP;
            case ')':
                return RP;
            case '+':
                return PLUS;
            case '-':
                return MINUS;
            case '*':
                return MULTIPLY;
            case '/':
                return DIVIDE;
            default:
                return UNKNOWN;
        }
    }
}

File: Assign2/RD_PY/Symbol.py
from enum import Enum


class Symbol(Enum):
    NUMBER = 1
    LP = 2
    RP = 3
    PLUS = 4
    MINUS = 5
    MULTIPLY = 6
    DIVIDE = 7
    UNKNOWN = 8

    def get_symbol(c):
        if (c == '0' or c == '1' or c == '2' or c == '3' or c == '4' or c == '5' or c == '6' or c == '7' or c == '8' or c == '9'):
            return Symbol.NUMBER
        elif (c == '('):
            return Symbol.LP
        elif (c == ')'):
            return Symbol.RP
        elif (c == '+'):
            return Symbol.PLUS
        elif (c == '-'):
            return Symbol.MINUS
        elif (c == '*'):
            return Symbol.MULTIPLY
        elif (c == '/'):
            return Symbol.DIVIDE
        else:
            return Symbol.UNKNOWN
